stdlib/solana.c

```diff
@@ -50,14 +43,9 @@ entrypoint(const uint8_t *input)
         }
     }
 
-    if (params.ka_cur != 0)
-    {
-        return ERROR_INVALID_INSTRUCTION_DATA;
-    }
-
     __init_heap();
 
-    return solang_dispatch(&params);
+    return solang_dispatch(params.input, params.input_len, &params);
```

Comment:
You are already receiving a pointer to `params`. Why are you passing `input` and `input_len` as separate arguments?